core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -466,40 +479,35 @@ public String getName() {
   }
 
   @Override
-  public void processJournalEntry(JournalEntry entry) throws IOException {
-    if (mDirectoryIdGenerator.replayJournalEntryFromJournal(entry)
-        || mInodeTree.replayJournalEntryFromJournal(entry)
-        || mMountTable.replayJournalEntryFromJournal(entry)
-        || mUfsManager.replayJournalEntryFromJournal(entry)
-        || mSyncManager.replayJournalEntryFromJournal(entry)) {
-      return;
-    } else {
-      throw new IOException(ExceptionMessage.UNEXPECTED_JOURNAL_ENTRY.getMessage(entry));
+  public boolean processJournalEntry(JournalEntry entry) {
+    for (Journaled journaled : mJournaledComponents) {
+      if (journaled.processJournalEntry(entry)) {
+        return true;
+      }
     }
+    return false;
   }
 
-  private void updateTxIdFromJournalEntry(File.ActiveSyncTxIdEntry activeSyncTxId) {
-    mSyncManager.setTxId(activeSyncTxId.getMountId(), activeSyncTxId.getTxId());
+  @Override
+  public void resetState() {
+    mJournaledComponents.forEach(Journaled::resetState);
   }
 
   @Override
-  public void resetState() {
-    mInodeTree.reset();
-    mMountTable.reset();
-    mSyncManager.reset();
+  public void writeToCheckpoint(OutputStream output) throws IOException, InterruptedException {
+    JournalUtils.writeToCheckpoint(output, mJournaledComponents);
+  }
+
+  @Override
+  public void restoreFromCheckpoint(InputStream input) throws IOException {
+    JournalUtils.restoreFromCheckpoint(input, mJournaledComponents);
   }
 
   @Override
   public Iterator<JournalEntry> getJournalEntryIterator() {
-    return Iterators.concat(mInodeTree.getJournalEntryIterator(),
-        mDirectoryIdGenerator.getJournalEntryIterator(),
-        // The mount table should be written to the checkpoint after the inodes are written, so that
```

Comment:
can we keep a version of this comment, for when we create the list, so we know what the ordering dependencies are?